lib/sps/test/src/SpsTest.cpp

```diff
@@ -0,0 +1,370 @@
+#include "SpsTest.h"
```

Comment:
each class should have its own test class.
e.g. UserInput should be in UserInputTest etc.